src/SFA.DAS.EmployerFinance/Models/LevyDeclarationSaga.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using SFA.DAS.EmployerFinance.Messages.Events;
+
+namespace SFA.DAS.EmployerFinance.Models
+{
+    public class LevyDeclarationSaga : Entity
+    {
+        public int Id { get; private set; }
+        public LevyDeclarationSagaType Type { get; private set; }
+        public DateTime PayrollPeriod { get; private set; }
+        public long HighWaterMarkId { get; private set; }
+        public DateTime Created { get; private set; }
+        public DateTime? Updated { get; private set; }
+        public int ImportPayeSchemeLevyDeclarationsTasksCount { get; private set; }
+        public int ImportPayeSchemeLevyDeclarationsTasksCompleteCount { get; private set; }
+        public DateTime? ImportPayeSchemeLevyDeclarationsTasksFinished { get; private set; }
```

Comment:
We could encapsulate the trio of properties Count / CompleteCount / TasksFinished into a class, and possibly include behaviour in the class also.

Reply:
Hmm I'm not sure how to map child objects to the same table as the parent with EF, probably as some sort of value object?

Reply:
It's probably not worth the extra complication in EF to do this!